node/pkg/common/guardianset.go

```diff
@@ -66,9 +89,15 @@ func (g *GuardianSet) KeysAsHexStrings() []string {
 // KeyIndex returns a given address index from the guardian set. Returns (-1, false)
 // if the address wasn't found and (addr, true) otherwise.
 func (g *GuardianSet) KeyIndex(addr common.Address) (int, bool) {
-	for n, k := range g.Keys {
-		if k == addr {
-			return n, true
+	if g.keyMap != nil {
+		if idx, found := g.keyMap[addr]; found {
+			return idx, true
+		}
+	} else {
+		for n, k := range g.Keys {
+			if k == addr {
+				return n, true
+			}
```

Comment:
Your code is totally reasonable here, but I always try to do as few indents as feasible to make code a bit easier to read. If you agree, something like this would work:

```suggestion
	if g.keyMap != nil {
		if idx, found := g.keyMap[addr]; found {
			return idx, true
		}
		return -1, false
	}
	for n, k := range g.Keys {
		if k == addr {
			return n, true
		}
```